### PR TITLE
Add support for enableHttp2 in prometheus remotewrite

### PR DESCRIPTION
**Pull request by hervenicol**
## Description

Add support for enable_http2 in prometheus remotewrite definitions.

see also https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write

Inspired by what was done for podmonitors and servicemonitors (#4889).

## Type of change

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

Add support for enableHttp2 in remotewrite definitions (in prometheus and prometheusagent CRDs)

```release-note
[ENHANCEMENT] Added support for enable_http2 in prometheus remotewrite configuration.
```

### CODE REVIEW

File: pkg/apis/monitoring/v1/prometheus_types.go [hunk 1]
```diff
@@ -1256,6 +1256,9 @@ type RemoteWriteSpec struct {
 	// MetadataConfig configures the sending of series metadata to the remote storage.
 	// +optional
 	MetadataConfig *MetadataConfig `json:"metadataConfig,omitempty"`
+
+	// Whether to enable HTTP2.
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/prometheus_types.go:1260]:
```suggestion
	// Whether to enable HTTP2.
	// +optional
```

**Review by simonpasquier — COMMENTED** [review]
one small nit otherwise LGTM

File: pkg/apis/monitoring/v1/prometheus_types.go (hunk 1 above)
Comment (hervenicol) [reply pkg/apis/monitoring/v1/prometheus_types.go:1260]: :+1: 
I just added it and re-generated the doc. :heavy_check_mark:

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**